tests/e2e/tests/add-google-sheets-datasource-flow.spec.ts

```diff
@@ -0,0 +1,256 @@
+import { type Page, test, expect } from '@playwright/test';
+import { performInitialSetup, navigateToSettings, navigateToDataSourceForm } from '../helpers/setup';
+import { getDataSourceNameInput, getGoogleSheetsUrlInput, getCreateButton } from '../helpers/selectors';
+
+test.describe('Add Google Sheets Data Source Flow', () => {
+  test.beforeEach(async ({ page }) => {
+    test.setTimeout(180000); // 3 minutes for OAuth complexity
+    await performInitialSetup(page);
+    await navigateToSettings(page);
+  });
+
+  test('Create Google Sheets data source with public spreadsheet URL', async ({ page }: { page: Page }) => {
+    await navigateToDataSourceForm(page, 'google-sheets');
+
+    const dbNameInput = getDataSourceNameInput(page);
+    await dbNameInput.waitFor({ state: 'visible', timeout: 10000 });
+    await dbNameInput.fill('test-google-sheets');
+
+    // Look for Google Sheets specific inputs - URL or spreadsheet ID
+    const urlInput = getGoogleSheetsUrlInput(page);
+
+    const urlInputExists = await urlInput.isVisible({ timeout: 5000 }).catch(() => false);
+
+    if (urlInputExists) {
+      // Use a simple public Google Sheets document that doesn't require authentication
+      await urlInput.fill(
+        'https://docs.google.com/spreadsheets/d/1BxiMVs0XRA5nFMdKvBdBZjgmUUqptlbs74OgvE2upms/edit?usp=sharing',
+      );
+    }
+
+    // Look for OAuth connect button or Google authentication
+    const connectButton = page
+      .locator('[data-testid="google-oauth-connect"]')
+      .or(
+        page.getByRole('button', { name: /connect to google|authenticate|sign in with google|connect google sheets/i }),
+      )
+      .or(page.locator('button:has-text("Connect to Google"), button:has-text("Authenticate")'));
+
+    const connectButtonExists = await connectButton.isVisible({ timeout: 5000 }).catch(() => false);
+
+    if (connectButtonExists) {
+      // Note: In a real test environment, we would mock the OAuth flow
+      await connectButton.click();
+      await page.waitForLoadState('networkidle');
+    }
+
+    // Use the shared selector function with extended fallbacks for Google Sheets
+    const saveButton = getCreateButton(page)
+      .or(page.locator('button:has-text("Create")'))
+      .or(page.locator('button:has-text("Save")'))
```

Comment:
Do we really need all the options and this way of selecting?
We could just have a single line with `data-testid` ?